src/Network/Network/AzureFirewallPolicy/SetAzureFirewallPolicyCommand.cs

```diff
@@ -0,0 +1,159 @@
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        [Parameter(
+                    Mandatory = true,
+                    ValueFromPipelineByPropertyName = true,
+                    HelpMessage = "The resource Id.", ParameterSetName = SetByResourceIdParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [SupportsWildcards]
+        public virtual string ResourceId { get; set; }
+
+        [Parameter(
+                    Mandatory = false,
+                    ValueFromPipelineByPropertyName = true,
+                    HelpMessage = "The operation mode for Threat Intelligence.")]
+        [ValidateSet(
+                    MNM.AzureFirewallThreatIntelMode.Alert,
+                    MNM.AzureFirewallThreatIntelMode.Deny,
+                    MNM.AzureFirewallThreatIntelMode.Off,
+                    IgnoreCase = false)]
+        public string ThreatIntelMode { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The base policy to inherit from")]
+        public string BasePolicy { get; set; }
+
+        [Parameter(
+                    Mandatory = true,
+                    ValueFromPipelineByPropertyName = true,
+                    HelpMessage = "location.", ParameterSetName = SetByNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [Parameter(Mandatory = true, ParameterSetName = SetByResourceIdParameterSet)]
+        public virtual string Location { get; set; }
+
+        [Parameter(
+                    Mandatory = false,
+                    ValueFromPipelineByPropertyName = true,
+                    HelpMessage = "A hashtable which represents resource tags.")]
+        public Hashtable Tag { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            if (this.IsParameterBound(c => c.ResourceId))
+            {
+                var resourceInfo = new ResourceIdentifier(ResourceId);
+                ResourceGroupName = resourceInfo.ResourceGroupName;
+                Name = resourceInfo.ResourceName;
+            }
+            else if (this.IsParameterBound(c => c.InputObject))
+            {
+                ResourceGroupName = InputObject.ResourceGroupName;
+                Name = InputObject.Name;
+                Location = InputObject.Location;
+            }
+
+            if (!NetworkBaseCmdlet.IsResourcePresent(() => GetAzureFirewallPolicy(ResourceGroupName, Name)))
+            {
+                throw new ArgumentException(Microsoft.Azure.Commands.Network.Properties.Resources.ResourceNotFound);
+            }
+
+            if (InputObject != null)
+            {
+                // Map to the sdk object
+                var firewallPolicy = NetworkResourceManagerProfile.Mapper.Map<MNM.FirewallPolicy>(InputObject);
```

Comment:
Other parameters that user specified should also be merged into `firewallPolicy` object -- consider this scenario:
```powershell
Get-AzFirewallPolicy -ResourceGroupName $rg -Name $name | Set-AzFirewallPolicy -Location "eastus"
```

Reply:
I am not sure I get this. Do you mind giving an example?

Reply:
Fixed it. I have verified passing the threat intelmode from the pipeline. We are not allowed to change the location for a policy once its created. so right now we would(backend service) throw an exception if we try to change it. However we need the location to update the policy or to call createOrUpdate function. So we cannot remove this field in case you are wondering to get rid of it entirely.